salt/utils/event.py

```diff
@@ -514,7 +516,10 @@ def destroy(self, linger=5000):
             # Wait at most 2.5 secs to send any remaining messages in the
             # socket or the context.term() below will hang indefinitely.
             # See https://github.com/zeromq/pyzmq/issues/102
-            self.sub.setsockopt(zmq.LINGER, linger)
+            try:
```

Comment:
or add a parameter to destroy to shallow or not shallow the exception. well, a matter of taste.